File: momentra/momentra/MomentraRadius.swift
//
//  MomentraRadius.swift
//  momentra
//
//  Created by santosh on 19/04/26.
//

import Foundation

/// Corner radius tokens — pill-heavy system (Momentra Theme Kit v2.2 HTML).
/// `.mc` moment demo uses 16pt; the kit table lists 14pt for generic home cards — use `card` for those, `momentCard` for goal/moment rows.
enum MomentraRadius {
    static let data: CGFloat = 4        // Data rows, list items
    static let input: CGFloat = 12      // Input fields
    static let cardSm: CGFloat = 12     // Small cards, activity
    static let card: CGFloat = 14       // Generic surfaces, bottom chrome
    static let momentCard: CGFloat = 16 // `.mc` moment row
    static let contextTabInner: CGFloat = 10 // Context switcher inner segment (`.ctx-btn`)
    static let button: CGFloat = 14     // CTA buttons
    static let hero: CGFloat = 18       // Hero cards, bottom sheets
    static let pill: CGFloat = 999      // Chips, tabs, badges, avatars
}

File: apk/app/src/main/java/app/momentra/ui/theme/MomentraRadius.kt
package app.momentra.ui.theme

import androidx.compose.ui.unit.dp

/**
 * Corner radius tokens — pill-heavy system ([momentra_themekit_v2.2.html]).
 * Note: HTML `.mc` moment demo uses 16px; the kit radius table lists 14dp for generic “home” cards — use [card] for those, [momentCard] for goal/moment rows.
 */
object MomentraRadius {
    val data = 4.dp
    val input = 12.dp
    val cardSm = 12.dp
    /** Generic elevated surfaces (demo-card, bottom chrome, sheets chrome). */
    val card = 14.dp
    /** Theme Kit `.mc` personal/group moment row corners. */
    val momentCard = 16.dp
    /** Inner context tab segment (`.ctx-btn` rounded rect, not full pill). */
    val contextTabInner = 10.dp
    val button = 14.dp
    val hero = 18.dp
    val pill = 999.dp
}
